src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php

```diff
@@ -211,6 +227,28 @@ public function iShouldNotSeeShipmentWithOrderNumber(OrderInterface $order): voi
         );
     }
 
+    /**
+     * @Then I should see :amount :product units in the list
+     */
+    public function iShouldSeeUnitsInTheList(int $amount, ProductInterface $product): void
+    {
+        $shipmentUnitsFromResponse = $this->responseChecker->getResponseContent($this->client->getLastResponse())['units'];
+
+        $productUnitsCounter = 0;
+        foreach ($shipmentUnitsFromResponse as $shipmentUnitFromResponse) {
+            /** @var ShipmentUnitInterface $shipmentUnit */
+            $shipmentUnit = $this->shipmentUnitRepository->findOneBy(['id' => explode('/', $shipmentUnitFromResponse)[4]]);
```

Comment:
I believe it's not what we should expect from our response. We're able to check the details of the `ShipmentUnit` object but we should be able to check it in the API response 🚀 I see 2 options there:
- show the `ShipmentUnit` by IRI and check its product name
- embed `ShipmentUnit`s in the `Shipment` response

I would go with the first solution, at least for now, as there could be a lot of shipment units for each shipment - which will result in a **huge** response 😄

Reply:
Ok but i must have four clients: 1) shipment, 2) shipmentUnit, when I get shippable(id) from shipmentUnit, in my case it is variant, then I use third client for variant, I get variant, and when I will have variant, I get product id and next fourth client get product by id, then i will have productName :D For code reduction i used Repository, but I can change it if my way is no necessary :D